File: lib/models/westops/traffic_code.dart
/// A Road Traffic Act offence entry.
///
/// WestOps stores these as four parallel arrays in
/// `lib/modles/trafficCode.dart`; here they are joined into a single
/// immutable record because every consumer needs all four fields together.
class TrafficCode {
  final String code;
  final String offenceDescription;
  final double fineAmount;
  final int demeritPoints;
  final String legalSection;

  const TrafficCode({
    required this.code,
    required this.offenceDescription,
    required this.fineAmount,
    required this.demeritPoints,
    required this.legalSection,
  });
}

File: backend/JcfDocumentHub.Domain/WestOps/TrafficCode.cs
namespace JcfDocumentHub.Domain.WestOps;

public sealed class TrafficCode
{
    public string Id { get; set; } = string.Empty;

    public string Code { get; set; } = string.Empty;

    public string Section { get; set; } = string.Empty;

    public string Offence { get; set; } = string.Empty;

    public decimal FineAmount { get; set; }

    public int DemeritPoints { get; set; }

    public string Statute { get; set; } = string.Empty;
}
